master/internal/configpolicy/postgres_task_config_policy_intg_test.go

```diff
@@ -0,0 +1,509 @@
+//go:build integration
+// +build integration
+
+package configpolicy
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/etc"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/ptrs"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSetNTSCConfigPolicies(t *testing.T) {
+	ctx := context.Background()
+	require.NoError(t, etc.SetRootPath(db.RootFromDB))
+	pgDB, cleanup := db.MustResolveNewPostgresDatabase(t)
+	defer cleanup()
+	db.MustMigrateTestPostgres(t, pgDB, db.MigrationsFromDB)
+
+	user := db.RequireMockUser(t, pgDB)
+	workspaceIDs := []int32{}
+
+	defer func() {
+		if len(workspaceIDs) > 0 {
+			err := db.CleanupMockWorkspace(workspaceIDs)
+			if err != nil {
+				log.Errorf("error when cleaning up mock workspaces")
+			}
+		}
+
+	}()
+
+	tests := []struct {
+		name     string
+		ntscTCPs *model.NTSCTaskConfigPolicies
+		global   bool
+		err      *string
+	}{
+		{
+			"invalid user id",
+			&model.NTSCTaskConfigPolicies{
+				LastUpdatedBy:   -1,
+				WorkloadType:    model.NTSCType,
+				InvariantConfig: DefaultCommandConfig(),
+				Constraints:     DefaultConstraints(),
+			},
+			false,
+			ptrs.Ptr("violates foreign key constraint"),
+		},
+		{
+			"valid config no constraint",
+			&model.NTSCTaskConfigPolicies{
+				LastUpdatedBy:   user.ID,
+				LastUpdatedTime: time.Now().UTC(),
+				WorkloadType:    model.NTSCType,
+				InvariantConfig: DefaultCommandConfig(),
+				Constraints:     model.Constraints{},
+			},
+			false,
+			nil,
+		},
+		{
+			"valid constraint no config",
+			&model.NTSCTaskConfigPolicies{
+				LastUpdatedBy:   user.ID,
+				LastUpdatedTime: time.Now().UTC(),
+				WorkloadType:    model.NTSCType,
+				InvariantConfig: model.CommandConfig{},
+				Constraints:     DefaultConstraints(),
+			},
+			false,
+			nil,
+		},
+		{
+			"valid constraint valid config",
+			&model.NTSCTaskConfigPolicies{
+				LastUpdatedBy:   user.ID,
+				LastUpdatedTime: time.Now().UTC(),
+				WorkloadType:    model.NTSCType,
+				InvariantConfig: DefaultCommandConfig(),
+				Constraints:     DefaultConstraints(),
+			},
+			false,
+			nil,
+		},
+		{
+			"valid config no constraint",
```

Comment:
nit: there's another test with the same name above, but with different arguments

Reply:
Good catch! Removed the extra test case (it was the same test case with one of the struct fields populated with its default value)